File: SCHOOL/Budapest/advanced_combinatorics/hw1.py

def is_legal(arr):
    check = [0]*20
    for a in arr:
        for b in arr:
            if (b > a):
                index = b-a
                check[index-1] = 1
                check[(21-index)-1] = 1
    for c in check:
        if c == 0: return 0
    return 1


for a in range(1,21):
    for b in range(a+1,21):
        for c in range(b+1,21):
            for d in range(c+1,21):
                for e in range(d+1,21):
                    arr = [a,b,c,d,e]
                    if is_legal(arr) == 1:
                        print(arr)
#arr = [1,2,4,10]

File: SCHOOL/Budapest/advanced_combinatorics/hw1.c
#include<stdio.h>

int legal(int *list) {
    int i = 0;
    int j = 0;

    int check[12] = {0,0,0,0,0,0,0,0,0,0,0,0};
    printf("here2\n");
    for (i = 0; i < 3; i++) {
            printf("%d\n",i);
        for (j = i+1; j < 3; j++) {
            printf("%d,%d\n",i,j);
            int index = list[j]-list[i]-1;
            check[index] = 1;
            check[(13-index)] = 1;
        }
    }

    for (i = 0; i < 13; i++) {
        printf("%d\n",check[i]);
        //if (i == 0) return 0;
    }
    return 1;
}

int main() {
    int list[4] = {1,2,4};
    printf("here1\n");
    printf("%d\n",legal(list));
}

